File: src/test/java/models/Avis.java
package models;

import java.sql.Date;

public class Avis {
    private int id;
    private int note;
    private String commentaire;
    private Date dateAvis;
    private int activiteId;
    private int personneId;

    public Avis() {}

    public Avis(int id, int note, String commentaire, Date dateAvis, int activiteId, int personneId) {
        this.id = id;
        this.note = note;
        this.commentaire = commentaire;
        this.dateAvis = dateAvis;
        this.activiteId = activiteId;
        this.personneId = personneId;
    }

    public Avis(int note, String commentaire, Date dateAvis, int activiteId, int personneId) {
        this.note = note;
        this.commentaire = commentaire;
        this.dateAvis = dateAvis;
        this.activiteId = activiteId;
        this.personneId = personneId;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getNote() { return note; }
    public void setNote(int note) { this.note = note; }

    public String getCommentaire() { return commentaire; }
    public void setCommentaire(String commentaire) { this.commentaire = commentaire; }

    public Date getDateAvis() { return dateAvis; }
    public void setDateAvis(Date dateAvis) { this.dateAvis = dateAvis; }

    public int getActiviteId() { return activiteId; }
    public void setActiviteId(int activiteId) { this.activiteId = activiteId; }

    public int getPersonneId() { return personneId; }
    public void setPersonneId(int personneId) { this.personneId = personneId; }

    @Override
    public String toString() {
        return "Avis{" +
                "id=" + id +
                ", note=" + note +
                ", commentaire='" + commentaire + '\'' +
                ", dateAvis=" + dateAvis +
                ", activiteId=" + activiteId +
                ", personneId=" + personneId +
                '}';
    }
}

File: src/Entity/Avis.php
<?php

namespace App\Entity;

use App\Entity\Personne;
use App\Entity\Activite;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;

#[ORM\Entity]
#[UniqueEntity(
    fields: ['personne', 'activite'],
    message: 'Vous avez déjà ajouté un avis pour cette activité.'
)]
class Avis
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(type: 'integer')]
    #[Assert\NotNull(message: 'La note est obligatoire.')]
    #[Assert\Range(
        min: 1,
        max: 5,
        notInRangeMessage: 'La note doit être comprise entre {{ min }} et {{ max }}.'
    )]
    private ?int $note = null;

    #[ORM\Column(type: 'text')]
    #[Assert\NotBlank(message: 'Le commentaire est obligatoire.')]
    #[Assert\Length(
        min: 3,
        minMessage: 'Le commentaire doit contenir au moins {{ limit }} caractères.',
        max: 1000,
        maxMessage: 'Le commentaire ne doit pas dépasser {{ limit }} caractères.'
    )]
    private ?string $commentaire = null;

    #[ORM\Column(name: 'dateAvis', type: 'datetime')]
    private ?\DateTimeInterface $dateAvis = null;

    #[ORM\ManyToOne(targetEntity: Activite::class, inversedBy: 'aviss')]
    #[ORM\JoinColumn(name: 'activite_id', referencedColumnName: 'id', nullable: false, onDelete: 'CASCADE')]
    #[Assert\NotNull(message: 'L’activité est obligatoire.')]
    private ?Activite $activite = null;

    #[ORM\ManyToOne(targetEntity: Personne::class, inversedBy: 'aviss')]
    #[ORM\JoinColumn(name: 'personne_id', referencedColumnName: 'id', nullable: false, onDelete: 'CASCADE')]
    #[Assert\NotNull(message: 'La personne est obligatoire.')]
    private ?Personne $personne = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNote(): ?int
    {
        return $this->note;
    }

    public function setNote(?int $note): self
    {
        $this->note = $note;
        return $this;
    }

    public function getCommentaire(): ?string
    {
        return $this->commentaire;
    }

    public function setCommentaire(?string $commentaire): self
    {
        $this->commentaire = $commentaire;
        return $this;
    }

    public function getDateAvis(): ?\DateTimeInterface
    {
        return $this->dateAvis;
    }

    public function setDateAvis(?\DateTimeInterface $dateAvis): self
    {
        $this->dateAvis = $dateAvis;
        return $this;
    }

    public function getActivite(): ?Activite
    {
        return $this->activite;
    }

    public function setActivite(?Activite $activite): self
    {
        $this->activite = $activite;
        return $this;
    }

    public function getPersonne(): ?Personne
    {
        return $this->personne;
    }

    public function setPersonne(?Personne $personne): self
    {
        $this->personne = $personne;
        return $this;
    }
}